src/router.js

```diff
@@ -1,72 +1,64 @@
-export function router(options) {
-  return function(emit) {
-    return {
-      state: {
-        router: {}
-      },
-      actions: {
-        router: {
-          set: function(state, actions, data) {
-            return {
-              router: data
-            }
-          },
-          go: function(state, actions, path) {
-            if (location.pathname + location.search !== path) {
-              history.pushState({}, "", path)
-              actions.router.set({
-                path: path
-              })
-            }
-          }
-        }
-      },
-      events: {
-        load: function(state, actions) {
-          addEventListener("popstate", function() {
-            actions.router.set({})
-          })
-        },
-        render: function(state, actions, view) {
-          return view[
-            (state.router.index >= 0
-              ? state
-              : actions.router.set(
-                  emit("route", match(location.pathname, view))
-                )).router.index
-          ][1]
-        }
+import { h } from 'hyperapp'
+
+// router module
```

Comment:
@speier Could you please move `router`, `Router` and `Route` into their own files:

- `router.js`
- `Router.js`
- `Route.js`

On-topic: Router & Route is familiar coming from React, but what  alternatives do we have? Just curious.

Reply:
ok, separated to their respective files.

Yes, Router & Route are coming from the react ecosystem probably. Not sure about any valid alternatives.